### ISSUE TITLE
weakref.WeakMethod incorrectly typed

### ISSUE THREAD

**Issue by A5rocks** [python/typeshed#5624]
The example listed in [the Python docs](https://docs.python.org/3/library/weakref.html#weakref.WeakMethod) doesn't type check:
```py
import weakref

class C:
    def method(self) -> None:
        print("method called!")

c = C()

r = weakref.WeakMethod(c.method)  # E: Argument 1 to "WeakMethod" has incompatible type "Callable[[], None]"; expected "MethodType"
```

Not sure why though :^)

**Comment by JelleZijlstra** [comment]
Yes, the issue is that the argument type is typed as `MethodType`, but type checkers will generally infer some sort of `Callable` as the type of a method, not `MethodType`.

That probably makes `WeakMethod` basically unusable in typed code, so I'd be OK with changing its type annotations to require a Callable instead at some cost in safety.

**Comment by srittau** [comment]
I'm fine with changing typeshed, although ideally type checkers would understand `MethodType`. But since this is a very rarely useful feature, support for it would have very low priority.
